rust/crd/src/history.rs

```diff
@@ -232,6 +235,93 @@ impl SparkHistoryServer {
         )
         .context(InvalidProductConfigSnafu)
     }
+
+    pub fn merged_env(
+        &self,
+        s3logdir: &S3LogDir,
+        role_group_env_overrides: HashMap<String, String>,
+    ) -> Vec<EnvVar> {
+        // Maps env var name to env var object. This allows env_overrides to work
+        // as expected (i.e. users can override the env var value).
+        let mut vars: BTreeMap<String, EnvVar> = BTreeMap::new();
+        let role_env_overrides = &self.role().config.env_overrides;
+
+        // This env var prevents the history server from detaching itself from the
+        // start script because this leads to the Pod terminating immediately.
+        vars.insert(
+            "SPARK_NO_DAEMONIZE".to_string(),
+            EnvVar {
+                name: "SPARK_NO_DAEMONIZE".to_string(),
+                value: Some("true".into()),
+                value_from: None,
+            },
+        );
+        vars.insert(
+            "SPARK_DAEMON_CLASSPATH".to_string(),
+            EnvVar {
+                name: "SPARK_DAEMON_CLASSPATH".to_string(),
+                value: Some("/stackable/spark/extra-jars/*".into()),
+                value_from: None,
+            },
+        );
+
+        let mut history_opts = vec![
+        format!("-Dlog4j.configurationFile={VOLUME_MOUNT_PATH_LOG_CONFIG}/{LOG4J2_CONFIG_FILE}"),
+        format!(
+            "-Djava.security.properties={VOLUME_MOUNT_PATH_CONFIG}/{JVM_SECURITY_PROPERTIES_FILE}"
+        ),
+        format!("-javaagent:/stackable/jmx/jmx_prometheus_javaagent.jar={METRICS_PORT}:/stackable/jmx/config.yaml")
+    ];
+
+        // if TLS is enabled build truststore
+        if tlscerts::tls_secret_name(&s3logdir.bucket.connection).is_some() {
+            history_opts.extend(vec![
+                format!("-Djavax.net.ssl.trustStore={STACKABLE_TRUST_STORE}/truststore.p12"),
+                format!("-Djavax.net.ssl.trustStorePassword={STACKABLE_TLS_STORE_PASSWORD}"),
+                format!("-Djavax.net.ssl.trustStoreType=pkcs12"),
+            ]);
+        }
+
+        vars.insert(
+            "SPARK_HISTORY_OPTS".to_string(),
+            EnvVar {
+                name: "SPARK_HISTORY_OPTS".to_string(),
+                value: Some(history_opts.join(" ")),
+                value_from: None,
+            },
+        );
+
+        // apply the role overrides
+        let mut role_envs = role_env_overrides.iter().map(|env_var| {
```

Comment:
Thanks for the suggestion. I've gone for a named tuple pair here: [ad32cef](https://github.com/stackabletech/spark-k8s-operator/pull/451/commits/ad32cefc6f441c9b609e05b8e8592f1f5f1beddd)

Reply:
I saw the same thing but I figured I don't want to further annoy @adwk67 with Rust nitpicking xD

Reply:
I think nits are good for communicating/sharing thoughts, but we shouldn't require them to be fixed in the PR (else they aren't nits).

I'm generally happy to fix pedantic nits myself rather than require the author to do them.